molgenis-security/src/main/java/org/molgenis/security/MolgenisChangePasswordFilter.java

```diff
@@ -40,7 +40,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
 		if ((authentication != null) && authentication.isAuthenticated() && !authentication.getName()
 																						   .equals(ANONYMOUS_USERNAME)
-				&& !httpRequest.getRequestURI().toLowerCase().endsWith(CHANGE_PASSWORD_URI.toLowerCase()))
+				&& !httpRequest.getRequestURI().toLowerCase().equals(CHANGE_PASSWORD_URI.toLowerCase()))
```

Comment:
Since this is security stuff I would like to see a unit test for this code branch.